Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationResultDispatchingValidatorTest.cs

```diff
@@ -0,0 +1,174 @@
+using Remotion.Web.UI.Controls;
+
+namespace Remotion.ObjectBinding.Web.UnitTests.UI.Controls.Validation
+{
+  [TestFixture]
+  public class BocListValidationResultDispatchingValidatorTest : BocTest
+  {
+    [Test]
+    public void RefreshErrorMessage_WithoutValidationFailures_ReturnsMessageFromResourceManager ()
+    {
+      var validationFailureRepoStub = new Mock<IBocListValidationFailureRepository>();
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocList(true))
+          .Returns(Array.Empty<BocListValidationFailureWithLocationInformation>());
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true))
+          .Returns(Array.Empty<BocListValidationFailureWithLocationInformation>());
+
+      var error = "New error just arrived";
+      var resourceManagerStub = new Mock<IResourceManager>();
+      resourceManagerStub.Setup(_ => _.TryGetString("Remotion.ObjectBinding.Web.UI.Controls.BocList.HasValidationFailureText", out error)).Returns(true);
+
+      var bocListMock = new Mock<Control> { CallBase = true };
+      bocListMock.As<IBocList>().Setup(_ => _.ValidationFailureRepository).Returns(validationFailureRepoStub.Object);
+      bocListMock.As<IBocList>().Setup(_ => _.GetResourceManager()).Returns(resourceManagerStub.Object);
+      bocListMock.Object.ID = "someBocList";
+
+      var validator = new BocListValidationResultDispatchingValidator();
+      validator.ErrorMessage = "This should not be output";
+      validator.ControlToValidate = "someBocList";
+
+      NamingContainer.Controls.Add(bocListMock.Object);
+      NamingContainer.Controls.Add(validator);
+
+      ((IValidatorWithDynamicErrorMessage)validator).RefreshErrorMessage();
+
+      Assert.That(validator.ErrorMessage, Is.EqualTo("New error just arrived"));
+    }
+
+    [Test]
+    public void RefreshErrorMessage_WithBocListValidationFailures_ReturnsFailuresOfBocList ()
+    {
+      var validationFailureRepoStub = new Mock<IBocListValidationFailureRepository>();
+
+      var bocListMock = new Mock<Control> { CallBase = true };
+      bocListMock.As<IBocList>().Setup(_ => _.ValidationFailureRepository).Returns(validationFailureRepoStub.Object);
+      bocListMock.Object.ID = "someBocList";
+
+      var validationFailuresForBocList = new[]
+                                         {
+                                           BocListValidationFailureWithLocationInformation.CreateFailure(BusinessObjectValidationFailure.Create("Errors on BocList"))
+                                         };
+
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocList(true)).Returns(validationFailuresForBocList);
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true))
+          .Returns(Array.Empty<BocListValidationFailureWithLocationInformation>());
```

Comment:
```suggestion
      validationFailureRepoStub
          .Setup(_ => _.GetUnhandledValidationFailuresForBocList(true))
          .Returns(validationFailuresForBocList);
      validationFailureRepoStub
          .Setup(_ => _.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true))
          .Returns(Array.Empty<BocListValidationFailureWithLocationInformation>());
```
(both, because they're pretty symetrical already)